app/helpers/embed_player_helper.rb

```diff
@@ -37,59 +38,65 @@ def embed_player_episode_url(ep, type = nil, preview = false)
-    params[EMBED_PLAYER_SEASON] = options[:season]
-    params[EMBED_PLAYER_CATEGORY] = options[:category]
+    if options[:all_episodes] === "all"
+      params[EMBED_PLAYER_PLAYLIST] = "all"
+    end
 
-    # TODO: the height styling doesn't really work here in the way Play specifies how it needs to be for a playlist, so leaving this as a TODO for now.
-    # if options[:embed_player_type] == "card" || options[:embed_player_type] == "fixed_card"
-    #   params[EMBED_PLAYER_CARD] = "1"
-    # end
+    if options[:all_episodes] == "number" && options[:episode_number].to_i > 1
+      params[EMBED_PLAYER_PLAYLIST] = options[:episode_number]
+    end
 
-    embed_params(params)
-  end
+    if !options[:season].to_s.strip.empty? && options[:season].to_i > 0
+      params[EMBED_PLAYER_SEASON] = options[:season]
+    end
 
-  def embed_player_episode_iframe(ep, type = nil, preview = false)
-    src = embed_player_episode_url(ep, type, preview)
-    allow = "monetization"
+    if !options[:category].to_s.strip.empty?
+      params[EMBED_PLAYER_CATEGORY] = options[:category]
+    end
 
-    if type == "card"
-      # TODO: this is NOW working, but I'm not sure how helpful this is to a producer that wishes to embed it.
-      tag.iframe src: src, allow: allow, width: "100%", height: "700", style: "--aspect-ratio: 2/3; width: 100%;"
-    elsif type == "fixed_card"
-      tag.iframe src: src, allow: allow, width: "500", height: "700"
-    else
-      tag.iframe src: src, allow: allow, width: "100%", height: "200"
+    if options[:embed_player_type] == "card"
+      params[EMBED_PLAYER_CARD] = "1"
     end
+
+    embed_params(params, preview)
   end
 
-  def embed_player_podcast_iframe(podcast, options, preview = false)
-    src = embed_player_podcast_url(podcast, options, preview)
+  def embed_player_iframe(options)
+    src = ""
     allow = "monetization"
+    iframe_height = "200"
+    iframe_styles = ""
+    wrapper_styles = "line-height: 0;"
 
     if options[:embed_player_type] == "card"
-      # TODO: this is NOW working, but I'm not sure how helpful this is to a producer that wishes to embed it.
-      tag.iframe src: src, allow: allow, width: "100%", height: "700", style: "--aspect-ratio: 2/3; width: 100%;"
-    elsif options[:embed_player_type] == "fixed_card"
-      tag.iframe src: src, allow: allow, width: "500", height: "700"
-    else
-      tag.iframe src: src, allow: allow, width: "100%", height: "200"
+      wrapper_styles = "position: relative; height: 0; width: 100%; padding-top: calc(100% + #{iframe_height}px); line-height: 0;"
+      iframe_height = "100%"
+      iframe_styles = "position: absolute; inset: 0;"
+    end
+
+    tag.div style: wrapper_styles, data: {embed_preview_target: "embedIframeWrapper"} do
+      tag.iframe class: "bg-light", src: src, allow: allow, width: "100%", height: iframe_height, style: iframe_styles, data: {embed_preview_target: "embedIframe"}
```

Comment:
So I'm curious - why the big diff in this method?

It looks like you've structurally moved the generation of the iframe code to the frontend / stimulus.  Rather than generating the changes in ruby and reloading via turbo.  So I'm not entirely sure when most of these conditional branches in this method would get called.

Reply:
Yes. The embed URL and iframe HTML is generated by play now. This logic is basically used once to initialize the iframe and wrapper based on configured defaults.